File: packages/zoom/android/src/main/kotlin/com/testpress/flutter_zoom_meeting_sdk/FlutterZoomMeetingSdkPluginResponse.kt
package com.testpress.flutter_zoom_meeting_sdk

data class StandardZoomResponse(
    val isSuccess: Boolean,
    val message: String,
    val action: String,
    val params: Map<String, Any>? = null
) {
    init {
        actionLog(action, " - $isSuccess - $message - $params")
    }

    fun toMap(): Map<String, Any> {
        return mapOf(
            "platform" to "android",
            "isSuccess" to isSuccess,
            "message" to message,
            "action" to action,
            "params" to (params ?: emptyMap<String, Any>())
        )
    }
}

File: packages/zoom/ios/zoom/Sources/zoom/FlutterZoomMeetingSdkPluginResponse.swift
struct StandardZoomResponse {
    let isSuccess: Bool
    let message: String
    let action: String
    let params: [String: Any]

    func toDictionary() -> [String: Any] {
        return [
            "platform": "ios",
            "isSuccess": isSuccess,
            "message": message,
            "action": action,
            "params": params,
        ]
    }
}

struct StandardZoomEventResponse {
    let event: String
    let oriEvent: String
    let params: [String: Any]

    func toDictionary() -> [String: Any] {
        return [
            "platform": "ios",
            "event": event,
            "oriEvent": oriEvent,
            "params": params,
        ]
    }
}
